woocommerce/wc-admin
# Add settings page with excluded order statuses

Fixes #1252 

Sorry for the length of this PR; in hindsight it may have benefited from being broken down into a few different PRs.  This PR:

* Adds the Settings route and page.
* Adds a reusable `Setting` component.
* Adds settings to the wc-api to read and update settings specific to wc-admin.
* Hooks up the excluded order status settings to those used in the reports.
* Add custom settings to the `wcSettings` global.
* Adds extensibility to the `wcSettings` global and to the settings page.

### Accessibility


### Screenshots
<img width="1665" alt="screen shot 2019-01-23 at 1 56 16 pm" src="https://user-images.githubusercontent.com/10561050/51586057-ae912480-1f16-11e9-9bab-37036aba2e2a.png">

### Detailed test instructions:

1.  Add custom statuses to your WooCommerce order statuses.  You can use [this plugin](https://wordpress.org/plugins/custom-order-statuses-woocommerce/) if you don't already have statuses added.
2.  Open the new Settings page at `wp-admin/admin.php?page=wc-admin#/analytics/settings`.
3.  Check that statuses are shown and that any custom statuses are shown as well.
4.  Change excluded statuses and save.  Refresh and make sure that settings persist.
5.  Check that reports reflect the excluded statuses chosen.
6.  Use the "Reset Defaults" button and make sure the default excluded statuses are restored (cancelled, failed, and pending).

### Design
The design in #1252 differs from some of the default WooCommerce styles and those have in wc-admin. 
 After a slack discussion with @LevinMedia we are holding off for confirmation on changes to the following before making further changes:
* Checkboxes
* Buttons
* Section header
* Success notice after save (this will probably need to be a separate PR)
